api/apps/api/src/modules/scenarios/planning-units/scenario-planning-units.service.ts

```diff
@@ -28,18 +28,30 @@ export class ScenarioPlanningUnitsService {
     });
   }
 
-  async getByStatus(
+  async getByStatusSetByUser(
     scenarioId: string,
     lockStatus: LockStatus,
   ): Promise<ScenariosPlanningUnitGeoEntity[]> {
     return await this.puRepo.find({
       where: {
         scenarioId,
         lockStatus,
```

Comment:
would it not work (sorry, I'm the kind of knolling OCD person 🤷🏼 ) to do something like
```suggestion
        lockStatus: [LockStatus.LockedIn, LockStatus.LockedOut].includes(lockStatus) ? lockStatus : IsNull()
```
here and avoid using the new `getAvailablePUsSetByUser()` method? I don't remember what the TypeORM breaking change we faced at some point with null-related queries was: I think it was something related to `is null`, but I am not sure, maybe this suggestion is nonsense.

If it could actually work, I would still wrap it in a tiny method that does the `LockStatus`-to-db-values mapping, and I think we even have something like this, but perhaps just in terms of class-transformer in the entity 🤔

Reply:
According to docs, the IsNull() should work in fin options, however it keeps getting translated to '= NULL',
even avoiding ternary operator and leaving just `status: IsNull()` leads to this.. out of curiosity I also left just one where option (lockStatus: IsNull() ), still same result..
```
async getByStatusSetByUser(
    scenarioId: string,
    lockStatus: LockStatus,
  ): Promise<ScenariosPlanningUnitGeoEntity[]> {
    return await this.puRepo.find({
      where: {
        scenarioId,
        lockStatus: IsNull(),
        setByUser: true,
      },
    });
  }
```
`query: SELECT "ScenariosPlanningUnitGeoEntity"."id" AS "ScenariosPlanningUnitGeoEntity_id", "ScenariosPlanningUnitGeoEntity"."project_pu_id" AS "ScenariosPlanningUnitGeoEntity_project_pu_id", "ScenariosPlanningUnitGeoEntity"."scenario_id" AS "ScenariosPlanningUnitGeoEntity_scenario_id", "ScenariosPlanningUnitGeoEntity"."lockin_status" AS "ScenariosPlanningUnitGeoEntity_lockin_status", "ScenariosPlanningUnitGeoEntity"."lock_status_set_by_user" AS "ScenariosPlanningUnitGeoEntity_lock_status_set_by_user", "ScenariosPlanningUnitGeoEntity"."xloc" AS "ScenariosPlanningUnitGeoEntity_xloc", "ScenariosPlanningUnitGeoEntity"."yloc" AS "ScenariosPlanningUnitGeoEntity_yloc", "ScenariosPlanningUnitGeoEntity"."protected_area" AS "ScenariosPlanningUnitGeoEntity_protected_area", "ScenariosPlanningUnitGeoEntity"."protected_by_default" AS "ScenariosPlanningUnitGeoEntity_protected_by_default", "ScenariosPlanningUnitGeoEntity"."feature_list" AS "ScenariosPlanningUnitGeoEntity_feature_list" FROM "scenarios_pu_data" "ScenariosPlanningUnitGeoEntity" WHERE ("ScenariosPlanningUnitGeoEntity"."scenario_id" = $1 **AND "ScenariosPlanningUnitGeoEntity"."lockin_status" = $2** AND "ScenariosPlanningUnitGeoEntity"."lock_status_set_by_user" = $3) -- PARAMETERS: ["32cedb9f-1a9a-4278-84f2-a79f28454fd4",null,true]`